src/test/java/org/benetech/servicenet/service/impl/ImportServiceConfidentialityImplIntTest.java

```diff
@@ -109,142 +172,113 @@ public void shouldNotCreateLocationIfConfidential() {
 
-        assertEquals(dbSize, locationService.findAll().size());
-        assertNull(updated);
+        assertEquals(1, physicalAddressService.findAll().size());
+        assertEquals(EXISTING_STRING, physicalAddressService.findAll().get(0).getAddress1());
     }
 
     @Test
     @Transactional
     public void shouldNotCreatePostalAddressIfConfidential() {
-        Location location = generateExistingLocation();
-        PostalAddress address = new PostalAddress().address1(NEW_STRING).city(NEW_STRING).stateProvince(NEW_STRING)
-            .location(location);
+        PostalAddress address = new PostalAddress().address1(NEW_STRING).city(NEW_STRING).stateProvince(NEW_STRING);
         address.setIsConfidential(true);
+        Location location = generateExistingLocation().postalAddress(address);
+        locationImportService.createOrUpdateLocation(location, EXISTING_EXTERNAL_ID, PROVIDER);
 
-        int dbSize = postalAddressService.findAll().size();
-        importService.createOrUpdatePostalAddress(address, location);
-
-        assertEquals(dbSize, postalAddressService.findAll().size());
+        assertEquals(0, postalAddressService.findAll().size());
     }
 
     @Test
     @Transactional
     public void shouldNotUpdatePostalAddressIfConfidential() {
-        Location location = generateExistingLocation();
-        PostalAddress existingAddress = new PostalAddress().address1(EXISTING_STRING).city(EXISTING_STRING).stateProvince(EXISTING_STRING)
-            .location(location);
-        em.persist(existingAddress);
-        em.flush();
-        em.refresh(location);
-
-        PostalAddress newAddress = new PostalAddress().address1(NEW_STRING).city(NEW_STRING).stateProvince(NEW_STRING)
-            .location(location);
-        newAddress.setIsConfidential(true);
+        PostalAddress existingAddress = new PostalAddress().address1(EXISTING_STRING).city(EXISTING_STRING).stateProvince(EXISTING_STRING);
+        Location location = generateExistingLocation().postalAddress(existingAddress);
+        locationImportService.createOrUpdateLocation(location, EXISTING_EXTERNAL_ID, PROVIDER);
+        assertEquals(1, postalAddressService.findAll().size());
 
-        int dbSize = locationService.findAll().size();
-        PostalAddress updated = importService.createOrUpdatePostalAddress(newAddress, location);
+        PostalAddress newAddress = new PostalAddress().address1(NEW_STRING).city(NEW_STRING).stateProvince(NEW_STRING);
+        newAddress.setIsConfidential(true);
+        location.setPostalAddress(newAddress);
+        locationImportService.createOrUpdateLocation(location, EXISTING_EXTERNAL_ID, PROVIDER);
 
-        assertEquals(dbSize, locationService.findAll().size());
-        assertNull(updated);
+        assertEquals(1, postalAddressService.findAll().size());
+        assertEquals(EXISTING_STRING, postalAddressService.findAll().get(0).getAddress1());
```

Comment:
In my opinion, this is misleading because the postal address is outdated.

Reply:
same here. I agree, but we probably have to create a separate a task for it, as it's working the same way as before changing the ImportService.